table.cc

```diff
@@ -180,7 +180,11 @@ table::make_reader(schema_ptr s,
         readers.emplace_back(mt->make_flat_reader(s, permit, range, slice, pc, trace_state, fwd, fwd_mr));
     }
 
-    if (cache_enabled() && !slice.options.contains(query::partition_slice::option::bypass_cache)) {
+    if (cache_enabled() && !slice.options.contains(query::partition_slice::option::bypass_cache)
+            && !slice.options.contains(query::partition_slice::option::reversed)) {
+        // FIXME: reenable cache for reversed reads after:
+        // - support for reversed reads is implemented in the cache,
+        // - fast forwarding is implemented in reversed sstable readers.
```

Comment:
See https://github.com/scylladb/scylla/pull/9174#discussion_r711722720

Reply:
I added an option to disable the new method for sstables, falling back to the old one but with cache enabled.
By default I enabled the new method so users are encouraged to switch and all integration/system tests automatically use the new method (no need to change the config). If users really see a significant performance regression for their reversed queries, they can use the fallback.
If you think a better default would be to use the old method then inform me and I'll change it.

Reply:
Given that the change can happen online, I'm okay with default on. We'll have to provide a good explanation in the release notes so people can choose to change the default on upgrade, and know what to watch for. Agree that getting the new code in use by default is good.